### PR TITLE
FAT-2586 new

### PR DESCRIPTION
**Pull request by TetianaParanich**
***
https://issues.folio.org/browse/FAT-2586
***
https://foliotest.testrail.io/index.php?/cases/view/343284
***

### CODE REVIEW

**Review by MaksimTerekhov — APPROVED** [review]

File: cypress/integration/data-import/make-fields-required-in-invoice-map-profile.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      permissions.viewOrganization.gui
+    ])
+      .then(userProperties => {
+        user = userProperties;
+
+        cy.login(user.username, user.password, { path: TopMenu.mappingProfilesPath, waiter: FieldMappingProfiles.waitLoading });
+      });
+  });
+
+  //   after(() => {
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/integration/data-import/make-fields-required-in-invoice-map-profile.js:28]: Done

File: cypress/integration/data-import/make-fields-required-in-invoice-map-profile.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
+import permissions from '../../support/dictionary/permissions';
+import TestTypes from '../../support/dictionary/testTypes';
+import DevTeams from '../../support/dictionary/devTeams';
+import TopMenu from '../../support/fragments/topMenu';
+import FieldMappingProfiles from '../../support/fragments/data_import/mapping_profiles/fieldMappingProfiles';
+import NewMappingProfile from '../../support/fragments/data_import/mapping_profiles/newMappingProfile';
+import Helper from '../../support/fragments/finance/financeHelper';
+import Users from '../../support/fragments/users/users';
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/integration/data-import/make-fields-required-in-invoice-map-profile.js]: This import is used in the after block. After remove comments it is used.

File: cypress/support/fragments/data_import/mapping_profiles/mappingProfileDetails.js [hunk 3]
```diff
@@ -67,5 +78,10 @@ export default {
           .click());
       }
     ));
+  },
+
+  checkErrorMessageIsPresented:(textFieldName) => {
+    cy.do(TextField(textFieldName).click());
+    cy.expect(TextField(including(textFieldName)).has({ error: 'Please enter a value' }));
```
Comment (penfriend) [inline cypress/support/fragments/data_import/mapping_profiles/mappingProfileDetails.js:85]: you can work with this object TextField(textFieldName) without using "including" and move it to const variable in this method

File: cypress/support/fragments/data_import/mapping_profiles/mappingProfileDetails.js [hunk 4]
```diff
@@ -57,6 +60,14 @@ export default {
     ]);
   },
 
+  deleteMappingProfile:(name) => {
+    cy.do([
+      Pane({ id:'full-screen-view' }).find(actionsButton).click(),
```
Comment (penfriend) [inline cypress/support/fragments/data_import/mapping_profiles/mappingProfileDetails.js:65]: Pane({ id:'full-screen-view' }) move to const var

File: cypress/support/fragments/data_import/mapping_profiles/newMappingProfile.js [hunk 5]
```diff
@@ -64,6 +66,10 @@ const selectOrganizationByName = (organizationName) => {
   ]);
 };
 
+const selectFromResultsList = (rowNumber = 0) => {
+  cy.do(Modal('Select Organization').find(MultiColumnListRow({ index: rowNumber })).click());
```
Comment (penfriend) [inline cypress/support/fragments/data_import/mapping_profiles/newMappingProfile.js:70]: please use the variable above
const organizationModal = Modal('Select Organization');

File: cypress/integration/data-import/match-on-pol-and-update-items.spec.js [hunk 6]
```diff
@@ -81,23 +81,23 @@ describe('ui-data-import: Match on POL and update related Instance, Holdings, It
...
     {
       mappingProfile: { typeValue: NewMappingProfile.folioRecordTypeValue.item,
         name: mappingProfileNameForItem,
         update: true },
       actionProfile: { typeValue: NewActionProfile.folioRecordTypeValue.item,
         name: actionProfileNameForItem,
-        action: 'Update (all record types except Orders, Invoices, or MARC Holdings, Invoices, or MARC Holdings)' }
+        action: 'Update (all record types except Orders, Invoices, or MARC Holdings)' }
     }
   ];
```
Comment (penfriend) [inline cypress/integration/data-import/match-on-pol-and-update-items.spec.js:103]: 'BasicOrderLine' is not defined

File: cypress/integration/data-import/match-on-pol-and-update-items.spec.js [hunk 7]
```diff
@@ -81,23 +81,23 @@ describe('ui-data-import: Match on POL and update related Instance, Holdings, It
...
       actionProfile: { typeValue: NewActionProfile.folioRecordTypeValue.instance,
         name: actionProfileNameForInstance,
-        action: 'Update (all record types except Orders, Invoices, or MARC Holdings, Invoices, or MARC Holdings)' }
+        action: 'Update (all record types except Orders, Invoices, or MARC Holdings)' }
     },
     {
       mappingProfile: { typeValue: NewMappingProfile.folioRecordTypeValue.holdings,
         name: mappingProfileNameForHoldings,
         update: true },
       actionProfile: { typeValue: NewActionProfile.folioRecordTypeValue.holdings,
```
Comment (penfriend) [inline cypress/integration/data-import/match-on-pol-and-update-items.spec.js:90]: Are you sure that the items you are searching in this test exist before test execution?

File: cypress/support/fragments/topMenu.js [hunk 8]
```diff
@@ -24,5 +24,6 @@ export default {
   checkInPath: '/checkin',
   checkOutPath: '/checkout',
   receivingPath: '/receiving',
-  bulkEditPath: '/bulk-edit'
+  bulkEditPath: '/bulk-edit',
+  mappingProfilesPath: '/settings/data-import/mapping-profiles'
```
Comment (penfriend) [inline cypress/support/fragments/topMenu.js:28]: we have already this path in the settingsMenu fragment. It is defined by the variable mappingProfilePath

File: cypress/support/fragments/topMenu.js [hunk 9]
```diff
@@ -24,5 +24,6 @@ export default {
   checkInPath: '/checkin',
   checkOutPath: '/checkout',
   receivingPath: '/receiving',
-  bulkEditPath: '/bulk-edit'
+  bulkEditPath: '/bulk-edit',
```
Comment (penfriend) [inline cypress/support/fragments/topMenu.js:27]: so editing this file is obsolete in this pull request

File: cypress/support/fragments/data_import/mapping_profiles/mappingProfileDetails.js (hunk 3 above)
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/mapping_profiles/mappingProfileDetails.js:85]: Done

File: cypress/support/fragments/data_import/mapping_profiles/mappingProfileDetails.js (hunk 4 above)
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/mapping_profiles/mappingProfileDetails.js:65]: Done

File: cypress/support/fragments/data_import/mapping_profiles/newMappingProfile.js (hunk 5 above)
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/mapping_profiles/newMappingProfile.js:70]: Done

File: cypress/integration/data-import/match-on-pol-and-update-items.spec.js (hunk 6 above)
Comment (TetianaParanich) [reply cypress/integration/data-import/match-on-pol-and-update-items.spec.js:103]: Done

File: cypress/integration/data-import/match-on-pol-and-update-items.spec.js (hunk 7 above)
Comment (TetianaParanich) [reply cypress/integration/data-import/match-on-pol-and-update-items.spec.js:90]: yes, it is created by creating an order

File: cypress/support/fragments/topMenu.js (hunk 8 above)
Comment (TetianaParanich) [reply cypress/support/fragments/topMenu.js:28]: Done

File: cypress/support/fragments/topMenu.js (hunk 9 above)
Comment (TetianaParanich) [reply cypress/support/fragments/topMenu.js:27]: Done

**Review by penfriend — APPROVED** [review]

**Pull request merged**